shield.go

```diff
@@ -17,6 +18,8 @@ const MaxLength = 4096
 // SaltLength is the length of the salt generated by GenerateSalt.
 const SaltLength = 32
 
+var ErrPasswordTooLong = errors.New("password too long")
```

Comment:
Because errors cannot  be constants: if you try to do`const err = errors.New("lorem")` the compiler will say `const initializer errors.New("lorem") is not a constant`. This makes sense, as constants are for [scalar values](https://blog.golang.org/constants), something that errors aren't.